here-naksha-app-service/src/test/java/com/here/naksha/app/service/SourceIdHandlerApiTest.java

```diff
@@ -70,6 +73,99 @@ void tc2001_testCreateFeaturesWithSourceIdInMetaWithTags() throws Exception {
                 .hasUuids();
 
     }
+    @Test
+    void tc2002_testCreateFeaturesWithoutSourceIdButWithSourceIdTag() throws Exception {
+
+        // Given:
+        final String bodyJson = loadFileOrFail("SourceHandlerId/TC2002_createFeatureWithoutSourceId/create_feature.json");
+        final String expectedBodyPart = loadFileOrFail("SourceHandlerId/TC2002_createFeatureWithoutSourceId/feature_response_part.json");
+        String streamId = UUID.randomUUID().toString();
+
+        // When
+        HttpResponse<String> response = getNakshaClient().post("hub/spaces/" + SPACE_ID + "/features", bodyJson, streamId);
+        // Then
+        assertThat(response)
+                .hasStatus(200)
+                .hasStreamIdHeader(streamId)
+                .hasJsonBody(expectedBodyPart, "Create Feature response body doesn't match")
+                .hasInsertedCountMatchingWithFeaturesInRequest(bodyJson)
+                .hasInsertedIdsMatchingFeatureIds(null)
+                .hasUuids();
+
+    }
+
+   @Test
+    void tc2003_searchBySourceId() throws Exception {
+
+        //given
+        final String bboxQueryParam = "west=-180&south=-90&east=180&north=90";
+        final String propQueryParam = "%s=eq=%s".formatted(
+                urlEncoded("p.@ns:com:here:mom:meta.sourceId"),
+                urlEncoded("task_2")
+        );
+
+        final String expectedBodyPart =
+                loadFileOrFail("SourceHandlerId/TC2003_searchBySourceId/feature_response_part.json");
+        String streamId = UUID.randomUUID().toString();
 
+        // When
+        HttpResponse<String> response = nakshaClient
+                .get("hub/spaces/" + SPACE_ID + "/bbox?" + bboxQueryParam + "&" + propQueryParam , streamId);
+
+        // Then
+        assertThat(response)
+                .hasStatus(200)
+                .hasStreamIdHeader(streamId)
+                .hasJsonBody(expectedBodyPart, "Get Feature response body doesn't match");
+    }
+
+    @Test
+    void tc2004_searchWithoutSourceId() throws Exception {
+
+        //given
+        final String bboxQueryParam = "west=-180&south=-90&east=180&north=90";
+        final String propQueryParam = "p.speedLimit='60'";
+
+        final String expectedBodyPart =
+                loadFileOrFail("SourceHandlerId/TC2004_searchWithoutSourceId/feature_response_part.json");
+        String streamId = UUID.randomUUID().toString();
+
+        // When
+        HttpResponse<String> response = nakshaClient
+                .get("hub/spaces/" + SPACE_ID + "/bbox?" + bboxQueryParam + "&" + propQueryParam , streamId);
+
+        System.out.println(response.body());
```

Comment:
delete this line please :)

Reply:
... I was checking three times did i removed it  :(